cmake/ActsCompilerOptions.cmake

```
@@ -2,10 +2,23 @@
 if (NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Build type configuration" FORCE)
   message(STATUS "Setting default build type: ${CMAKE_BUILD_TYPE}")
-endif() 
+endif()
+
+set(extra_flags "-Wall;-Wextra;-Wpedantic;-Wshadow;-Wno-unused-local-typedefs")
+
+# This adds some useful conversion checks like float-to-bool, float-to-int, etc.
+# However, at the moment this is only added to clang builds, since GCC's -Wfloat-conversion
+# is much more aggressive and also triggers on e.g., double-to-float
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|AppleClang")
```

I know this is how it was before as well, but why not:

```suggestion
if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

? If "AppleClang" matches, "Clang" will too.

---

Probably. The matching was there previously, but I can change it anyway.